### PR TITLE
feat(chrome)!: improves markup structure for `Nav` items

### PR DESCRIPTION
**Pull request by geotrev**
## Description

Introduces `Nav.List` subcomponent for a semantically accurate navigation list.

## Detail

- introduced `Nav.List` as a wrapper for `Nav.Item`
- React Context is used to detect if `Nav.Item` is within a `Nav.List`, and if so, includes a styled `li`
- Requires consumers to keep logo/brandmark items out of `Nav.List`

New composition API:

```jsx
<Chrome>
  <Nav> // <nav>
    <Nav.Item hasLogo /> // div
    <Nav.List> // <ul>
      <Nav.Item /> // <li>
      <Nav.Item />
      <Nav.Item />
    </Nav.List>
    <Nav.Item hasBrandmark /> // div
  </Nav>
</Chrome>
```

## Checklist

- ~~:ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)~~
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [x] :wheelchair: tested for WCAG 2.1 AA accessibility compliance
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

File: packages/chrome/src/styled/nav/StyledNavList.ts
Comment (ze-flo) [inline packages/chrome/src/styled/nav/StyledNavList.ts:20]: CSS `order` rules can lead to accessibility problems and I've had mixed experiences with it. Do we still need these rules or should we leave it to the user to specify the correct order (potential breaking change 💥)?

https://developer.mozilla.org/en-US/docs/Web/CSS/order#accessibility_concerns
https://www.w3.org/TR/css-flexbox-1/#order-accessibility

Comment (geotrev) [reply packages/chrome/src/styled/nav/StyledNavList.ts:20]: Actually, you just jogged my senses a bit. Since this element is just a `ul` and it's a container, it naturally shouldn't land in any tab/render order. So, we can probably just remove it I think. 🤔 

but your concern is valid and I don't think we need to include `order` in these elements since the original tab oder already prescribes the correct navigation behavior. I'll give it a quick test to confirm, and we can discuss inconsistencies if there are any!

Comment (geotrev) [reply packages/chrome/src/styled/nav/StyledNavList.ts:20]: I think I see why `order` is used here.

It's to ensure that a child composition of:

```jsx
<Nav>
  <Nav.List>
    <Nav.Item hasBrandmark />
    <Nav.Item>
      <Nav.ItemText />
      <Nav.ItemIcon />
    </Nav.Item>
    <Nav.Item hasLogo />
  </Nav.List>
</Nav>
```

Renders in the expected way in the DOM (using React components for simplicity):

```jsx
<nav>
  <ul>
    <li class="StyledLogoItem />
    <li class="StyledItem">
      <span class="StyledItemIcon" />
      <span class="StyledItemText" />
    </li>
    <li class="StyledBrandmarkItem" />
  </ul>
</nav>
```

E.g., a break-prevention measure. I can't speak to if this is still desirable though. I assume so but maybe JZ or design have further insights. I would agree it is a bit heavy-handed but I can understand the logic after toying with it.

Comment (ze-flo) [reply packages/chrome/src/styled/nav/StyledNavList.ts]: Thanks for your excellent explanation. 💯  It looks like removing `order` isn't so trivial after all and definitely falls out of scope for this PR. Perhaps something we can revisit in the future.

Comment (geotrev) [reply packages/chrome/src/styled/nav/StyledNavList.ts]: Yeah, I agree on revisiting if it makes sense!

Comment (jzempel) [reply packages/chrome/src/styled/nav/StyledNavList.ts]: The `order` CSS throughout Chrome is intentional, and despite the a11y concerns we keep it because enforcing consistent UX tab ordering despite consumer usage is the more important concern. Perhaps if/when a grid layout vs. flex is explored in the future, we can reevaluate.

**Review by jzempel — CHANGES_REQUESTED** [review]
In accordance with the a11y issue we're attempting to address, we're looking for logo and brandmark items to live _outside_ the list. Expecting to see:

```jsx
<Nav>
  <Nav.Item hasLogo />
  <Nav.List>
    <Nav.Item ... />
    <Nav.Item ... />
    <Nav.Item ... />
  </Nav.List>
  <Nav.Item hasBrandmark />
</Nav>
```

This means `<Nav.Item>` will need context to understand whether to render a `<li>` or not.

**Comment by geotrev** [comment]
Thanks @jzempel - that detail wasn't immediately clear to me from our previous conversations. I recall you mentioned context but I didn't connect that to the logo/brandmark items being excluded from the list. Should be updated now.

**Review by jzempel — APPROVED** [review]
Approved, with minor URL comment for consideration.

File: docs/migration.md
Comment (geotrev) [reply docs/migration.md:80]: Oh, I wasn't sure if that was possible here. Good call. I'll update and test it before I merge!

**Comment by geotrev** [comment]
Dang @ze-flo amazing snipe on that pattern file! haha

**Pull request merged**
